pkg/agents/team7/agent1/agent.go

```diff
@@ -79,99 +93,539 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
+		a.SendMessage(msg2)
+		a.opMem.msg3Sent = true
+	}
+
+}
+
+func (a *CustomAgent7) manageNewDay() {
 
-	//eat
+	a.opMem.seenPlatform = false
+	a.opMem.prevHP = a.HP()
+	a.opMem.foodEaten = 0
+	a.opMem.receivedReq = false
 
-	//choses random number between -5 and 5
-	if a.CurrPlatFood() != -1 {
+}
 
-		r1 := rand.Intn(11) - 5
-		r2 := rand.Intn(11) - 5
+func (a *CustomAgent7) treatyGT(tActive messages.Treaty, foodtotake food.FoodType, available food.FoodType, amount food.FoodType) food.FoodType {
+
+	if a.HP() > tActive.ConditionValue() || int(a.CurrPlatFood()) > tActive.ConditionValue() {
+		switch tActive.RequestOp() {
+		case messages.GT:
+			if foodtotake > available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount) - 1
+			}
+		case messages.GE:
+			if foodtotake >= available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		case messages.EQ:
+			if foodtotake != available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		}
+	}
+	return foodtotake
 
-		a.kindness += (a.personality.neuroticism * r1) / 50
-		a.greediness += (a.personality.neuroticism * r2) / 50
+}
+func (a *CustomAgent7) treatyGE(tActive messages.Treaty, foodtotake food.FoodType, available food.FoodType, amount food.FoodType) food.FoodType {
 
-		if a.kindness < 0 {
-			a.kindness = 0
-		} else if a.kindness > 100 {
-			a.kindness = 100
+	if a.HP() >= tActive.ConditionValue() || int(a.CurrPlatFood()) >= tActive.ConditionValue() {
+		switch tActive.RequestOp() {
+		case messages.GT:
+			if foodtotake > available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount) - 1
+			}
+		case messages.GE:
+			if foodtotake >= available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		case messages.EQ:
+			if foodtotake != available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
 		}
+	}
+	return foodtotake
+
+}
+func (a *CustomAgent7) treatyEQ(tActive messages.Treaty, foodtotake food.FoodType, available food.FoodType, amount food.FoodType) food.FoodType {
 
-		if a.greediness < 0 {
-			a.greediness = 0
-		} else if a.greediness > 100 {
-			a.greediness = 100
+	if a.HP() == tActive.ConditionValue() || int(a.CurrPlatFood()) == tActive.ConditionValue() || a.Floor() == tActive.ConditionValue() {
+		switch tActive.RequestOp() {
+		case messages.GT:
+			if foodtotake > available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount) - 1
+			}
+		case messages.GE:
+			if foodtotake >= available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		case messages.EQ:
+			if foodtotake != available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
 		}
+	}
```

Comment:
You've copied this same block of code five times - you can put this into a function.